crates/katana/core/src/service/messaging/starknet.rs

```diff
@@ -61,15 +61,14 @@ impl StarknetMessaging {
         })
     }
 
-    /// Fetches events for the given blocks range.
     pub async fn fetch_events(
         &self,
         from_block: BlockId,
         to_block: BlockId,
-    ) -> Result<HashMap<u64, Vec<EmittedEvent>>> {
+    ) -> Result<Vec<EmittedEvent>> {
         trace!(target: LOG_TARGET, from_block = ?from_block, to_block = ?to_block, "Fetching logs.");
 
-        let mut block_to_events: HashMap<u64, Vec<EmittedEvent>> = HashMap::new();
+        let mut events: Vec<EmittedEvent> = Vec::new();
```

Comment:
```suggestion
        let mut events = vec![];
```
Usually, if the compiler inference is possible, we choose to use it. When functions are simple, it makes the statements easier to read.